extensions/amp-embedly-card/1.0/component.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '#preact';
+import {MessageType, ProxyIframeEmbed} from '#preact/component/3p-frame';
+import {deserializeMessage} from '../../../src/3p-frame-messaging';
+import {forwardRef} from '#preact/compat';
+import {useCallback, useState} from '#preact';
+
+/**
+ * Attribute name used to set api key with name
+ * expected by embedly.
+ * @const {string}
+ */
+const API_KEY_ATTR_NAME = 'data-card-key';
+
+const FULL_HEIGHT = '100%';
+
+/**
+ * @param {!EmbedlyCardDef.Props} props
+ * @param {{current: ?EmbedlyCardDef.Api}} ref
+ * @return {PreactDef.Renderable}
+ */
+export function EmbedlyCardWithRef(
+  {requestResize, style, title, url, ...rest},
+  ref
+) {
+  const [height, setHeight] = useState(null);
+  const messageHandler = useCallback(
+    (event) => {
+      const data = deserializeMessage(event.data);
+      if (data['type'] == MessageType.EMBED_SIZE) {
+        const height = data['height'];
+        if (requestResize) {
+          requestResize(height);
+          setHeight(FULL_HEIGHT);
+        } else {
+          setHeight(height);
+        }
+      }
+    },
+    [requestResize]
+  );
+
+  // Check for valid props
+  if (!checkProps(url)) {
+    return null;
+  }
+
+  // Prepare options for ProxyIframeEmbed
+  const iframeOptions = {
+    url,
+  };
+
+  // Extract Embedly Key
+  const ampEmbedlyKeyElement = document.querySelector('amp-embedly-key');
+  const apiKey = ampEmbedlyKeyElement?.getAttribute('value');
```

Comment:
It is preferred to avoid imperative DOM APIs in the Preact layer whenever possible. Fortunately the concept of a global, once-provided "EmbedlyKey" sounds to me like a [Context value in React](https://reactjs.org/docs/context.html). Let's do the following:

(1) In the Preact layer, create an additional `EmbedlyContext` in a new file `embedly-context.js`:
```js
import {createContext} from '#preact';

const EmbedlyContext = createContext(
  /** @type {EmbedlyCardDef.EmbedlyContext} */ ({apiKey})
);
export {EmbedlyContext};
```
(2) Have the EmbedlyCard component here consume the context.
```js
import {EmbedlyContext} from './embedly-context'

const {apiKey} = useContext(EmbedlyContext);
```

(3) In the AMP layer, wrap the component with the context provider:
```js
function EmbedlyCardWithContext(props) {
  const ampEmbedlyKeyElement = document.querySelector(KEY_TAG);
  const apiKey = ampEmbedlyKeyElement?.getAttribute('value');
  return (
    <EmbedlyContext.Provider value={apiKey}>
      <EmbedlyCard {...props}></EmbedlyCard>
    </EmbedlyContext.Provider>
  );
}

BaseElement['Component'] = EmbedlyCardWithContext
```

(4) Update Storybooks accordingly, with `Basic.amp.js` adding a sample that uses `<amp-embedly-key>` and `Basic.js` adding a sample that uses `<EmbedlyContext.Provider>`.